src/centos/scripts/start-marklogic.sh

```diff
@@ -147,6 +147,49 @@ function restart_check {
     error "Failed to restart $1" exit
 }
 
+################################################################
+# setting default protocol and CA cert for curl calls to bootstrap host
+################################################################
+ML_BOOTSTRAP_PROTOCOL="http"
+ML_CACERT_FILE="/run/secrets/${MARKLOGIC_JOIN_CACERT_FILE}"
+
+################################################################
+# Check if TLS is enabled on Bootstrap host
+################################################################
+function validate_tls_parameters {
+    local bootstrap_status=$1 host_protocol
+    host_protocol=$(get_host_protocol "${MARKLOGIC_BOOTSTRAP_HOST}")
+    if [[ "${bootstrap_status}" != "localhost" ]]; then
+        if [[ "${MARKLOGIC_JOIN_TLS_ENABLED}" == "true" ]]; then
+            if [[ -f "${ML_CACERT_FILE}" ]] && [[ -n "$(<"${ML_CACERT_FILE}")" ]]; then
+                if [[ "${host_protocol}" == "https" ]]; then
+                    ML_BOOTSTRAP_PROTOCOL="https"
+                    # validate the CA certificate
+                    validate_cert "${ML_CACERT_FILE}"
+                    info "MARKLOGIC_JOIN_TLS_ENABLED and MARKLOGIC_JOIN_CACERT_FILE are set, TLS will be used for joining cluster."
+                else
+                    error "TLS is not enabled on bootstrap_host_name host, please verify the configuration. Container shutting down." exit
+                fi
+            else
+                error "MARKLOGIC_JOIN_CACERT_FILE is not set, please review the configuration. Container shutting down." exit
+            fi
+        elif [[ "${MARKLOGIC_JOIN_TLS_ENABLED}" == "false" ]] || [[ -z "${MARKLOGIC_JOIN_TLS_ENABLED}" ]] || [[ -v "${MARKLOGIC_JOIN_TLS_ENABLED}" ]]; then
+            if [[ "${host_protocol}" == "https" ]]; then
+                error "TLS is enabled on ${MARKLOGIC_BOOTSTRAP_HOST}, please verify the configuration. Container shutting down." exit
+            else
+                info "MARKLOGIC_JOIN_TLS_ENABLED is not set, using HTTP for joining cluster."
+            fi
+        else
+            error "MARKLOGIC_JOIN_TLS_ENABLED must be set to true or false, please review the configuration. Container shutting down." exit
+        fi
+    fi
+}
+
+function validate_cert {
+    local cacertfile=$1
+    resp=$(curl -s -S -L --cacert "${cacertfile}" --ssl "${ML_BOOTSTRAP_PROTOCOL}"://"${MARKLOGIC_BOOTSTRAP_HOST}":8001 --anyauth --user "${ML_ADMIN_USERNAME}":"${ML_ADMIN_PASSWORD}")
```

Comment:
the resp is not handle. Should it be printed out?

Reply:
this curl call should only print/return errors in case of SSL issues. Hence printing $resp is not needed here. This has been tested by passing an invalid certificate in the parameter.

Reply:
updated fn to exit with error